src/audio/module_adapter/module_adapter.c

```diff
@@ -677,6 +677,8 @@ int module_adapter_copy(struct comp_dev *dev)
 	}
 
 	if (mod->simple_copy) {
+		if (!source_c)
+			return -ENODATA;
```

Comment:
Aa, so this is basicly to keep static checkers happy? You do need to "goto out" in this case to free the buffers on error.

Reply:
It is my first time using that tool and I thought that it would catch such trivial mistake.

Reply:
That's the thing with static analysis tools -- they lack context on what is correct and thus they only help you not crash/get other forms of UB, but NOT actually write the correct code when doing so.

Reply:
But this is invalid report, see:
```
if (mod->simple_copy) { // line 612
...
	source_c = buffer_acquire(source);
...
} else {
...
}

if (mod->simple_copy) { // line 679, here
	// source_c can not be NULL.
}